### PR TITLE
Add media:thumbnail and itunes:image to generated rss feeds

### PR DESCRIPTION
**Pull request by orthur**
PR for #64

### CODE REVIEW

File: app/api/server.go [hunk 1]
```diff
@@ -146,7 +148,14 @@ func (s *Server) getFeedCtrl(w http.ResponseWriter, r *http.Request) {
 
 	// replace link to UI page
 	if s.Conf.System.BaseURL != "" {
-		rss.Link = s.Conf.System.BaseURL + "/feed/" + feedName
+		baseURL := s.Conf.System.BaseURL
+		rss.Link = baseURL + "/feed/" + feedName
+		rss.ItunesImage = feed.ItunesImg{
+			URL: baseURL + "/image/" + feedName,
```
Comment (umputun) [inline app/api/server.go:154]: we shouldn't form this image URL automatically but rather conditionally from s.Conf.Feeds[feedName].Image (if not empty)

Comment (orthur) [reply app/api/server.go]: got it, fixed

Comment (umputun) [reply app/api/server.go]: almost ;) the image url should be `baseURL+s.Conf.Feeds[feedName].Image` instead of generic `baseURL + "/image/" + feedName`

**Comment by orthur** [comment]
Ok will add

**Comment by umputun** [comment]
in fact this server wasn't too testable. I have added ctx param to Run so the integration test could terminate it and extracted router() func, so you could pass it into httptest.Server

regarding the test itself - all you need to start it to make [a testing boltdb](https://github.com/umputun/feed-master/blob/master/app/youtube/service_test.go#L38-L43) and config filled with whatever you need and pass it in. 

After this, you can call bolt's save for several records and should be good to go.

so, your test would look like:

```go
  	tmpfile := filepath.Join(os.TempDir(), "test.db")
	defer os.Remove(tmpfile)
	db, err := bolt.Open(tmpfile, 0o600, &bolt.Options{Timeout: 1 * time.Second})
	require.NoError(t, err)
	boltStore := &proc.BoltDB{DB: db}

        conf := config.Conf{.....}
        srv := Server{Store: boltStore, Conf: conf}
        ts := httptest.Server(srv.router())
        defer  ts.Close()
```

From this point, you could send the real http request to ts.URL + "/whatever" and check the response

hope it helps

**Comment by orthur** [comment]
that helps, thanks

**Comment by umputun** [comment]
wanna me to merge this pr as-is and you could add test(s) in the following pr?

**Comment by orthur** [comment]
yes lets do that

**Pull request merged**